compiler/vertex-gen.py

```diff
@@ -54,11 +54,11 @@ def output_include_directive(f, name):
         f.write('#include "%s/vertex-%s.h"\n' % (REL_DIR, name))
 
 
-def output_class_header(f, base_name, name):
+def output_class_header(f, base_name, name, final=""):
     f.write("""
 template<>
-class vertex_inner<{name}> : public vertex_inner<{base_name}> {{
-public:""".format(name=name, base_name=base_name))
+class vertex_inner<{name}> {final}: public vertex_inner<{base_name}> {{
```

Comment:
Maybe, `final_specifier`, the same as below?
(`final` seemed to be boolean at first)

Reply:
ok